### PR TITLE
Make Cython work without recipe

### PR DESCRIPTION
**Pull request by JonasT**
**What this does:** This merge request makes Cython work for the non-recipe regular build. This means CythonRecipe is now unnecessary if the project doesn't need Android-specific extra changes. That's good because a recipe with no actual content is redundant, and because CythonRecipe is problematic:

**Why CythonRecipe is problematic:** I noticed CythonRecipe results in the packages own `setup.py` not run for Cythonization, and instead, p4a engages some manual Cython compilation which is broken per design for a couple of reasons:

1. it assumes all `.pyx` files anywhere in the project dir must compile. This breaks projects that have any `.pyx` files that are *not meant* to be built for the main build and break when Cythonized out of the blue
2. it uses the wrong `site-packages` because it uses host python. This breaks `.pxd` cross-includes from any dependencies, since this requires proper site-packages during Cythonize (this affected me, see #1473 )
3. it assumes the `setup.py` won't produce new `.pyx` files on its own. This breaks projects where that happens

(Please note this pull request doesn't fix CythonRecipe in any way, it just makes it avoidable for Android-aware Cython projects)

**Needs quite some testing, obviously!**

### CODE REVIEW

**Comment by JonasT** [comment]
Side note: I'm unhappy about how this pull request makes things *based* on CythonRecipe for the env vars of the regular build, since I think the CythonRecipe should be a special case of special needs Cython projects, not the other way around. However, I didn't take care of this yet because I wanted to first see if this works for other projects at all, and then gradually move stuff out of CythonRecipe and into the core later, stripping down CythonRecipe to its special uses in that process.

(Also, it would move quite a lot of stuff around, so just for cleanliness I think it'd be nicer to have it in a separate pull request later)

**Comment by tito** [comment]
About your 1. just to let you know Kivy source code have pyx that doesn't work and required on Android (like window_x11.pyx, etc). So maybe you should have an option to give a list of files to ignore?

**Comment by JonasT** [comment]
> About your 1. just to let you know Kivy source code have pyx that doesn't work and required on Android (like window_x11.pyx, etc). So maybe you should have an option to give a list of files to ignore?

That's like a hack of another hack. The real problem is that the `setup.py` already does this for any sane project, and p4a's CythonRecipe basically assumes the `setup.py` never works for any project for Cythonizing which IMHO is just a bad assumption. (It would be ok as a manual per-recipe enabled workaround, but as a default behavior for Cython projects I don't think it makes a lot of sense)

Also, I don't have any non-compiling `.pyx` files either, I was just explaining why CythonRecipe kinda sucks.

**Comment by JonasT** [comment]
It appears to be working mostly fine, but I can't fully test since I'm currently stuck on #1487

**Comment by JonasT** [comment]
Okay, so I finally made it through an app build to test this, and while the build succeeds and the app is packaged with the Cython parts successfully in the site-packages, this happens when it is launched:

> ImportError: dlopen failed: cannot locate symbol "PyExc_DeprecationWarning" referenced by "/mnt/expand/50b713c2-bef0-4226-aea1-000678d55d07/user/0/com.example.myapp/files/app/_python_bundle/site-packages/wobblui/`__core__.so`"...

I wonder, since there is no ELF loading error, whether the architecture/toolchain worked out correctly, but the linked python version or headers are possibly wrong? Anyone having an idea what the problem could be?

**Comment by JonasT** [comment]
(this is still WIP though, I want to clean up some stuff - but I'd **love** to hear test results!! Even if your app doesn't use Cython, just to make sure other things aren't broken by this)

**Comment by JonasT** [comment]
Okay, I cleaned up the parts I still wanted to clean up.

**To test this and give me useful feedback, it is enough to just build your project with this merge request.** While it needs some adjustments to use the new functionality, the much more important question is *whether it breaks existing stuff.* It changes the regular non-recipe python module install env, so it has the potential to cause some interference, so some wider testing wouldn't be a bad idea.

So anyone who's interested in testing, please do and report back! It works fine for me is all I can say

**Comment by JonasT** [comment]
Just a tiny note: the latest force pushes were purely rebasing to latest master since I rely on this pull request for my own builds quite deeply, not actual changes. As I wrote above, **it is done and ready for testing.**

**Comment by JonasT** [comment]
@AndreMiras by the way, any clue on when we're going to merge this? It's pretty much been sitting here for two weeks, I use it for my active project and it's done as far as I'm concerned... so IMHO it would make sense to review & merge this *if* you guys want this feature. I personally think moving away from recipes where they aren't strictly necessary is always an improvement, but that's just my view on things

**Comment by inclement** [comment]
Thanks, sorry for the delay on this one.

File: pythonforandroid/recipe.py [hunk 1]
```diff
@@ -989,8 +981,15 @@ def build_cython_components(self, arch):
...
 
+            self.strip_object_files(arch, env)
+
+    def strip_object_files(self, arch, env, build_dir=None):
+        if build_dir is None:
+            build_dir = self.get_build_dir(arch.arch)
+        with current_directory(build_dir):
             info('Stripping object files')
             if self.ctx.python_recipe.name == 'python2legacy':
+                info('Stripping object files')
```
Comment (AndreMiras) [inline pythonforandroid/recipe.py]: Minor: isn't this redundant with the one two lines above?

**Comment by JonasT** [comment]
oops yeah, I had  a conflict there and must have messed that line during merge/rebase

**Comment by AndreMiras** [comment]
Well that's not a big deal anyway, very good work as usual and sorry for letting the PR rot for so long. Great that inclement took over and merged

**Pull request merged**